app/include/zmk/display.h

```diff
@@ -6,4 +6,48 @@
 
 #pragma once
 
-int zmk_display_init();
\ No newline at end of file
+struct k_work_q *zmk_display_work_q();
+
+bool zmk_display_is_initialized();
+int zmk_display_init();
+
+/**
+ * Macro to define a ZMK event listener that handles the thread safety of fetching
+ * the necessary state from the system work queue context, invoking a work callback
+ * in the display queue context, and properly accessing that state safely when performing
+ * display/LVGL updates.
+ * @param listener THe ZMK Event manager listener name
+ * @param state_type The struct/enum type used to store/transfer state.
+ * @param cb The callback to invoke in the dispaly queue context to update the UI
+ * @param state_func The callback function to invoke to fetch the updated state from ZMK core
+ **/
+#define ZMK_DISPLAY_WIDGET_LISTENER(listener, state_type, cb, state_func)                          \
+    K_MUTEX_DEFINE(listener##_mutex);                                                              \
+    void listener##_work_cb(struct k_work *work) {                                                 \
+        LOG_DBG("");                                                                               \
+        cb();                                                                                      \
```

Comment:
Would it make more sense to pass the state into the callback function? Then users of this macro wouldn't need to know about the (currently undocumented) `listener##_get_local_state` function.

Reply:
Indeed! Just pushed an updated that refactors that to make that a bit more "pure" and not have to worry about the locally generated functions.